util/pkg/reflectutils/access.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reflectutils
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func SetString(target interface{}, targetPath string, newValue string) error {
+	targetValue := reflect.ValueOf(target)
+
+	targetFieldPath, err := ParseFieldPath(targetPath)
+	if err != nil {
+		return fmt.Errorf("cannot parse field path %q: %w", targetPath, err)
+	}
+
+	visitor := func(path *FieldPath, field *reflect.StructField, v reflect.Value) error {
+		if !targetFieldPath.HasPrefixMatch(path) {
+			return nil
+		}
+
+		if targetFieldPath.Matches(path) {
+			if !v.CanSet() {
+				return fmt.Errorf("cannot set field %q (marked immutable)", path)
+			}
+
+			t := v.Type().String()
+
+			var newV reflect.Value
+
+			switch t {
+
+			case "string":
+				newV = reflect.ValueOf(newValue)
+
+			case "*string":
+				newV = reflect.ValueOf(&newValue)
+
+			case "bool", "*bool":
+				b, err := strconv.ParseBool(newValue)
+				if err != nil {
+					return fmt.Errorf("cannot interpret %q value as bool", newValue)
+				}
+				if t == "*bool" {
+					newV = reflect.ValueOf(&b)
+				} else {
+					newV = reflect.ValueOf(b)
+				}
+
+			case "int64", "*int64", "int32", "*int32", "int", "*int":
+				v, err := strconv.Atoi(newValue)
+				if err != nil {
+					return fmt.Errorf("cannot interpret %q value as integer", newValue)
+				}
+				if strings.HasPrefix(t, "*") {
+					newV = reflect.ValueOf(&v)
+				} else {
+					newV = reflect.ValueOf(v)
+				}
+
+			case "[]string":
+				newV = reflect.Append(v, reflect.ValueOf(newValue))
+
+			default:
+				// This handles enums and other simple conversions
```

Comment:
We could also support things that are convertible from int or bool. Also slices of things convertible from string, int, or bool.